onlinejudge/service/yukicoder.py

```diff
@@ -262,9 +262,9 @@ def download_system_cases(self, *, session: Optional[requests.Session] = None) -
         """
 
         session = session or utils.get_default_session()
-        if not self.get_service().is_logged_in(session=session):
+        if 'Authorization' not in session.headers and not self.get_service().is_logged_in(session=session):
             raise NotLoggedInError
-        url = 'https://yukicoder.me/problems/no/{}/testcase.zip'.format(self.problem_no)
+        url = '{}/testcase.zip'.format(self.get_url())
```

Comment:
```suggestion
        url = 'https://yukicoder.me/problems/no/{}/testcase.zip'.format(self.problem_no)
```

ここ意味的に嘘なので戻します
(id で指定されたときに動かないのはそうですが、この変更をしてもいずれにせよ動かないし、動かないことが分かりにくくなるだけなので不適切)

Reply:
これ手元で動いたんですが

Reply:
これというのは oj download https://yukicoder.me/problems/3281 -a -t {TOKEN} と oj download https://yukicoder.me/problems/no/880 -a -t {TOKEN} のことです

Reply:
もしかして `https://yukicoder.me/problems/no/PROBLEM_NO/testcase.zip` の形式と `https://yukicoder.me/problems/PROBLEM_ID/testcase.zip` の形式が両方存在していますか？
(だとすればバグ修正なのでよさそう。この修正は「token を使えるようにする」とまったく無関係なので、デバッグ中のゴミがコミットされたのかなと思った :bow:)

Reply:
念のため yukiさんに確認してみます　

Reply:
確認しました　両方存在します

Reply:
ありがとう。ではこれは普通にバグ修正なのでうれしい